sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -52,6 +52,8 @@ static void _override_node(const ipv6_addr_t *addr, unsigned iface,
                            _nib_onl_entry_t *node);
 static inline bool _node_unreachable(_nib_onl_entry_t *node);
 
+#define INDEX_OF(a, e) (((e) - (a)) / sizeof(*e))
```

Comment:
>  I'd rather prefer a dedicated `_nib_offl_entry_t` function for this for type-safety for this btw.

So actually what I would prefer would be something like

```suggestion
static inline unsigned _nib_offl_idx_of(_nib_offl_entry_t *array,
                                        _nib_offl_entry_t *entry)
{
    return (entry - array) / sizeof(*entry);
```